autoscheduler/frontend/src/tests/CourseSelectCard.test.tsx

```diff
@@ -65,3 +64,43 @@ test('Remembers state after collapse', async () => {
   // checks that the card remembers how many elements were checked
   expect(checked2).toEqual(checked1);
 });
+
+test('Changes sections in response to changing course', async () => {
+  // arrange
+  const nodeProps = Object.create(Node.prototype, {});
+  // @ts-ignore
+  document.createRange = (): Range => ({
+    setStart: (): void => {},
+    setEnd: (): void => {},
+    commonAncestorContainer: {
+      ...nodeProps,
+      nodeName: 'BODY',
+      ownerDocument: document,
+    },
+  });
+  const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+  const {
+    getByText, getByLabelText, getAllByText,
+  } = render(
+    <Provider store={store}><CourseSelectCard id={0} /></Provider>,
+  );
+
+  /* ACT */
+  // fill in course
+  const courseEntry = getByLabelText('Course') as HTMLInputElement;
+  act(() => { fireEvent.click(courseEntry); });
+  act(() => { fireEvent.click(getByText('CSCE 121')); });
+  // switch to section view
+  act(() => { fireEvent.click(getByText('Section')); });
+  const course1Sections = (await waitForElement(() => getAllByText(/50\d/))).length;
+
+  // change course and read sections again
+  act(() => { fireEvent.change(courseEntry, { target: { value: 'MATH 15' } }); });
```

Comment:
The reason this commit took so long is that initially, I was trying to use `fireEvent.click` and `fireEvent.keyPress` to simulate user input. However, in Jest, clicks and key presses don't actually fire a `change` event, and because the autocomplete is a controlled input, its value will not update until it receives a `change` event. Good to keep in mind if you're writing frontend tests in the future.